Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -283,10 +283,64 @@ CFTypeID CGPathGetTypeID() {
     return __CGPath::GetTypeID();
 }
 
+// A helper for determining the number of points per path element type
+static inline int CGPathGetExpectedPointCountForType(CGPathElementType type) {
+    return type == kCGPathElementMoveToPoint || type == kCGPathElementAddLineToPoint ?
+               1 :
+               type == kCGPathElementAddQuadCurveToPoint ? 2 : type == kCGPathElementAddCurveToPoint ? 3 : 0;
+}
+
+// Create a mimic of CGPathElement that holds the points in a vector with a convenient copy constructor.
+struct _CGPathElementVector {
+    CGPathElementType _type;
+    std::vector<CGPoint> _points;
+    _CGPathElementVector(const CGPathElement el)
```

Comment:
You'll want `const CGPathElement&`, otherwise the compiler will emit a copy of CGPathElement that will be immediately destroyed after you copy it again.